testing_and_setup/compass/ocean/scripts/vertical_grid/make_vertical_grid.py

```diff
@@ -0,0 +1,198 @@
+#!/usr/bin/env python
+'''
+This script creates the vertical grid for MPAS-Ocean and writes it to a netcdf file.
+'''
+# import modules
+# {{{
+from netCDF4 import Dataset
+import numpy as np
+import argparse
+# }}}
+
+
+def main():
+    # parser
+    # {{{
+    parser = argparse.ArgumentParser(
+        description=__doc__,
+        formatter_class=argparse.RawTextHelpFormatter)
+    parser.add_argument(
+        '-o', '--output_file_name', dest='output_filename_name',
+        default='MPAS-Ocean_vertical_grid.nc',
+        help='MPAS file name for output of vertical grid.',
+        metavar='NAME')
+    parser.add_argument(
+        '-p', '--plot_vertical_grid', dest='plot_vertical_grid',
+        action='store_true')
+    parser.add_argument(
+        '-bd', '--bottom_depth', dest='bottom_depth',
+        default=5000,
+        help='bottom depth for the chosen vertical coordinate [m]',
+        type=float)
+    parser.add_argument(
+        '-nz', '--num_vert_levels', dest='nz',
+        default=64,
+        help='Number of vertical levels for the grid',
+        type=int)
+    parser.add_argument(
+        '-dz1', '--layer1_thickness', dest='dz1',
+        default=2.0,
+        help='Target thickness of the first layer [m]',
+        type=float)
+    parser.add_argument(
+        '-dz2', '--maxLayer_thickness', dest='dz2',
+        default=250.0,
+        help='Target maximum thickness in column [m]',
+        type=float)
+    parser.add_argument(
+        '-eps', '--error_tolerance', dest='epsilon',
+        default=1e-2,
+        help='Threshold for iterations', type=float)
+    parser.add_argument(
+        '-maxit',
+        '--max_iterations',
+        dest='maxit',
+        default=1000,
+        help='maximum number of iterations for grid convergences',
+        type=int)
+    args = parser.parse_args()
+
+    create_vertical_grid(args.bottom_depth, args.nz, args.dz1,
+                         args.dz2, args.plot_vertical_grid,
+                         maxit=args.maxit, epsilon=args.epsilon,
+                         outFile=args.output_filename_name)
+# }}}
+
+
+def create_vertical_grid(
+        bottom_depth,
+        nz,
+        dz1_in,
+        dz2_in,
+        plot_vertical_grid,
+        maxit,
+        epsilon,
+        outFile):
```

Comment:
I would define all these arguments with the same default values as in the command-line interface so they are optional.
```
bottom_depth=5000.0,
nz=64,
...
```

Reply:
Also, there should be a docstring documenting this function. It can basically copy the help:
```
"""
This function creates the vertical grid for MPAS-Ocean and writes it to a NetCDF file.

Parameters
----------

bottom_depth : float
    bottom depth for the chosen vertical coordinate [m]

nz : int
    Number of vertical levels for the grid
...
"""
```